### PR TITLE
Refs #6916 Added log to IDL parser.

### PR DESCRIPTION
**Pull request by LuisGP**
This PR adds the logging feature to the parser, including:
- Implementation
- Testing
- Documentation

A potential example could be implemented along with a parser example. A standalone `Log` example doesn't make sense to me.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/46/)

**Comment by LuisGP** [comment]
The Jenkins job failed because it is expecting the *not yet merged* generator-parser roundtrip test.

File: README.md [hunk 1]
```diff
@@ -495,6 +497,19 @@ It has, two helper functions to ease the retrieval of all types:
...
+The parser comes with a `Log` utility. The parser will log automatically different kind of events happened while
+parsing an IDL. Using the `Context` the user can customize the `Log` behavior and retrieve its results:
+
+- `log_level(LogLevel)` This method sets up the verbosity level (or severity) of the log.
+The available options are `DEBUG`, `INFO`, `WARNING`, and `ERROR` (by default `WARNING`).
+- `log_level()` Retrieves the current verbosity level.
+- `print_log(enable)` When enabled, it prints each log event to the standard output when parsing.
+- `log()` Retrieves the complete list of LogEvent created during the parsing.
+- `log(LogLevel, strict)` Retrieves a subset of LogEvent created during the parsing filtering by LogLevel.
+If strict is set to true, then it will return **only** LogEvents with the same LogLevel. Else (default),
```
Comment (richiware) [inline README.md]:
```suggestion
If `strict` is set to true, then it will return **only** LogEvents with the same LogLevel. Else (default),
```

**Review by richiware — CHANGES_REQUESTED** [review]

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/54/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/55/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/56/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/57/)

**Review by richiware — APPROVED** [review]

**Pull request merged**